### PR TITLE
feat: TreeView primitive

### PR DESCRIPTION
**Pull request by MildTomato**
## What kind of change does this PR introduce?

- adds `<Treeview/>` component
- added stories
- `<TreeViewItem/>` also supports editing state
- `<ContextMenu/>` exposed
- `<ContextMenu/>` classnames updated to use our colors
- no changes to `./apps/**`
  - should be no impact in prod, possible that package changes have impact though

https://github.com/supabase/supabase/assets/8291514/15274b70-0be2-4a01-8ebf-1fe6cdd395e1

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: packages/ui/src/components/TreeView/TreeView.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+      ...props
+    },
+    ref
+  ) => {
+    const [localValueState, setLocalValueState] = useState(name)
+    const inputRef = useRef<HTMLInputElement>(null)
+
+    useEffect(() => {
+      const handleClickOutside = (event: MouseEvent) => {
+        console.log('handleClickOutside')
```
Comment (joshenlim) [inline packages/ui/src/components/TreeView/TreeView.tsx:52]: remove console log

File: packages/ui/src/components/TreeView/TreeView.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      document.addEventListener('mousedown', handleClickOutside)
+
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside)
+      }
+    }, [localValueState])
+
+    useEffect(() => {
+      if (isEditing) {
+        console.log('useEffect with inputRef.current?.focus()')
```
Comment (joshenlim) [inline packages/ui/src/components/TreeView/TreeView.tsx:67]: remove console log

File: packages/ui/src/components/TreeView/TreeView.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                'group-aria-selected:text-foreground-light',
+                'group-aria-expanded:text-foreground-light'
+              )}
+              isOpen={isExpanded}
+              size={16}
+              strokeWidth={1.5}
+            />
+          </>
+        ) : (
+          <SQL_ICON
```
Comment (joshenlim) [inline packages/ui/src/components/TreeView/TreeView.tsx:148]: should we have the icon be passed in as a prop? make it extensible
can default to SQL_ICON if none provided

File: packages/ui/src/components/TreeView/TreeView.tsx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+            )}
+            size={16}
+            strokeWidth={1.5}
+          />
+        )}
+        <span className={cn(isEditing && 'hidden')}>{name}</span>
+        <form autoFocus onSubmit={handleSubmit} className={cn(!isEditing && 'hidden')}>
+          <Input
+            ref={inputRef}
+            onFocus={(e) => {
```
Comment (joshenlim) [inline packages/ui/src/components/TreeView/TreeView.tsx:164]: do we need this prop?

File: packages/ui/src/components/TreeView/index.tsx
Comment (joshenlim) [inline packages/ui/src/components/TreeView/index.tsx:1]: delete this file?

File: packages/ui/src/components/TreeView/TreeView.tsx (hunk 3 above)
Comment (MildTomato) [reply packages/ui/src/components/TreeView/TreeView.tsx:148]: yea sure

File: packages/ui/src/components/TreeView/TreeView.tsx (hunk 4 above)
Comment (MildTomato) [reply packages/ui/src/components/TreeView/TreeView.tsx]: nope, removed

File: packages/ui/src/components/TreeView/TreeView.tsx (hunk 3 above)
Comment (MildTomato) [reply packages/ui/src/components/TreeView/TreeView.tsx:146]: still pending to change by me

**Review by joshenlim — APPROVED** [review]

**Pull request merged**